app/models/backup_code_configuration.rb

```diff
@@ -36,7 +39,29 @@ def find_with_code(code:, user_id:)
       return if code.blank?
       code = code.downcase.strip
       code_fingerprint = create_fingerprint(code)
-      find_by(code_fingerprint: code_fingerprint, user_id: user_id)
+
+      user_salt_costs = select(:code_salt, :code_cost).
+        distinct.
+        where(user_id: user_id).
+        where.not(code_salt: nil, code_cost: nil).
+        limit(BackupCodeGenerator::NUMBER_OF_CODES * 2).
```

Comment:
```suggestion
      user_salt_costs = select(:code_salt, :code_cost).
        where(user_id: user_id).
        where.not(code_salt: nil).where.not(code_cost: nil)
```

Based on above suggestions, and `replaces where.not(code_salt: nil, code_cost: nil)` which does `WHERE NOT (code_salt IS NULL AND code_cost IS NULL)` that could include rows where only one or the other is null when we only want rows where both are not null. It's unlikely based on our implementation, but may as well.

Reply:
ah yup yup thanks, I forgot that `where.not` has funny semantics like this